backend/lib/database.ts

```diff
@@ -46,14 +55,36 @@ export class DatabaseQueries {
       await DatabaseQueries.db
         .update(pixels)
         .set({ color })
-        .where(eq(pixels.id, `${x},${y}`));
+        .where(
+          and(
+            eq(pixels.canvas, DatabaseQueries.activeCanvas),
+            eq(pixels.x, x),
+            eq(pixels.y, y)
+          )
+        );
     }
   }
+
   async setup() {
     const queries = [
-      `CREATE TABLE IF NOT EXISTS ${DATABASE_NAME} (id TEXT PRIMARY KEY, x INTEGER, y INTEGER, color TEXT)`,
+      "CREATE TABLE IF NOT EXISTS `pixels` (`canvas` INTEGER, `x` INTEGER, `y` INTEGER, `color` INTEGER, PRIMARY KEY(`canvas`, `x`, `y`))",
+      "CREATE TABLE IF NOT EXISTS `canvases` (`id` INTEGER NOT NULL, `name` TEXT, PRIMARY KEY(`id`))",
     ] as any;
 
-    await DatabaseQueries.db.run(sql(queries as TemplateStringsArray));
+    DatabaseQueries.db.run(sql(queries as TemplateStringsArray));
+
+    if (!INITIAL_CANVAS) {
+      throw new Error("INITIAL_CANVAS is a required env variable.");
+    } else {
+      const canvas = await DatabaseQueries.db
+        .select()
+        .from(canvases)
+        .where(eq(canvases.name, INITIAL_CANVAS));
+
+      if (canvas.length < 1)
```

Comment:
just my personal formatting quirks I'm usually against 
```js
if (a)
    then b;
```
if the whole thing can't be written as 
```js
if (a) then b
```
in one line; I prefer
```js
if (a) {
    then b;
}
```
for clarity, but since I don't have a CONTRIBUTING.md my point is moot lol